src/guides/v2.3/get-started/gs-curl.md

```diff
@@ -28,6 +28,128 @@ Option | Description
 `-T` | Transfers the specified local file to the remote URL.
 `-X` | Specifies the request method to use when communicating with the HTTP server. The specified request method is used instead of the default GET method.
 
+### Using cUrl in Magento
+
+Magento provides its own service-wrapper for using cURL instead of using default php cURL. The class ``Magento\Framework\HTTP\Client\Curl`` can be used to work with HTTP protocol using cURL library.
+At first, the instance of ``Magento\Framework\HTTP\Client\Curl`` should be created.
+
+```php
+/**
+* Constructor.
+*
+* @param Magento\Framework\HTTP\Client\Curl $curl
+*/
+public function __construct(
+   Magento\Framework\HTTP\Client\Curl $curl
+) {
+   $this->curl = $curl;
+}
+```
+#### Make GET request using cURL
+
+```php
+// get method
+$this->curl->get($url);
+
+// output of curl request
+$result = $this->curl->getBody();
+```
+
+Where the ``$url`` is the endpoint url
+
+#### Make POST request using cURL
+
+```php
+// post method
+$this->curl->post($url, $params);
+
+// output of curl requestt
+$result = $this->curl->getBody();
+```
+
+Where the ``$url`` is the endpoint url, ``$params`` is an array of data that is being sent via the POST request, the extra parameters can be added in the url.
+``$params`` example:
+
+```php
+$params = [
+  'user[email]' => $user->getEmail(),
+  'user[cellphone]' => $providerInfo['phone_number'],
+  'user[country_code]' => $providerInfo['country_code'],
+]
+```
+
+The Curl client can also adds headers, basic authorization, additional cURL options and cookies in the curl request. The Curl client provides these methods before using ``get`` or ``post`` method.
+
+#### Set curl header using addHeader method
+
+The ``addHeader`` method accepts two parameters. The curl header name and a curl header value.
+
+```php
+$this->curl->addHeader("Content-Type", "application/json");
+$this->curl->addHeader("Content-Length", 200);
+```
+
+#### Set curl header using setHeaders method
+
+The ``setHeaders`` method accepts an array as a parameter.
+
+```php
+$headers = ["Content-Type" => "application/json", "Content-Length" => "200"];
+$this->curl->setHeaders($headers);
+```
+
+#### Set basic authorization in Curl
+Set the basic authorization using the ``setCredentials`` method.
+
+```php
+$userName = "User_Name";
+$password = "User_Password";
+
+$this->curl->setCredentials($userName, $password);
+```
+
+It is equivalent to setting CURLOPT_HTTPHEADER value
+
+```php
+"Authorization : ". "Basic ".base64_encode($userName.":".$password)
+```
+
+#### Set curl option using setOption method
+
+The ``setOption`` method accepts two parameters. The curl option name and the cURL option value.
```

Comment:
```suggestion
The ``setOption`` method accepts two parameters. The cURL option name and the cURL option value.
```

Reply:
Please, make sure that we use the same spelling for the entire document. So we don't have Curl, curl, cURL.

Thank you.